dali/kernels/imgproc/color_manipulation/equalize/hist.cu

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stdexcept>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "dali/kernels/imgproc/color_manipulation/equalize/hist.h"
+
+namespace dali {
+namespace kernels {
+namespace equalize {
+namespace hist {
+
+__global__ void ZeroMem(const SampleDesc *sample_descs) {
+  auto sample_desc = sample_descs[blockIdx.y];
+  sample_desc.out[blockIdx.x * SampleDesc::range_size + threadIdx.x] = 0;
+}
```

Comment:
is this better than cudaMemsetAsync?

Reply:
I didn't try it, likely not.

Reply:
For single sample it seems to perform sligtly slower, but it is such a slim difference that I am not sure if that is a real thing. One concern is that I'd have to assume the tensor list is contigious here (or make num_sample calls).